kata-containers: Set cluster-stability to Disruptive

ldoktor opened the pull request:
Kata containers are a bit heavy for testing clusters, allow monitor failures by setting --cluster-stability to Disruptive (functional test issues are still reported, only the background health-checks are ignored)

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
OK so it is supported only on 4.14 and even there it still reports the failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44073/rehearse-44073-periodic-ci-kata-containers-kata-containers-main-next-e2e-tests/1710897512149159936

I'll dig deeper to find out more

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse periodic-ci-kata-containers-kata-containers-main-next-e2e-tests

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
/pj-rehearse

ldoktor commented:
@wainersm I think it's ready. This version points to pastebin as we don't have the wrapper merged just yet. Once it's reviewed and tested I will modify the last commit to point to https://raw.githubusercontent.com/openshift/release/master/ci-operator/config/kata-containers/kata-containers/wrapped-openshift-tests.sh so logically the CI is going to fail then, but we'll have the results from pastebin run so we should be able to force-push and therefore to create the link.

wainersm commented on ci-operator/config/kata-containers/kata-containers/wrapped-openshift-tests.sh:
```diff
@@ -0,0 +1,26 @@
+
+echo "$OUT"
+
+[ "$RET" -eq 0 ] || exit 0
```
@ldoktor shouldn't it be '&&' instead of '||'?

ldoktor replied:
Oh, yep, good catch

ldoktor commented on ci-operator/config/kata-containers/kata-containers/kata-containers-kata-containers-main__next.yaml:
```diff
@@ -15,6 +20,14 @@ images:
+    FROM tests-base
+    RUN mv /usr/bin/openshift-tests /usr/bin/openshift-tests-original
+    ADD https://pastebin.com/raw/Qkw03Fg1 /
+    RUN cat /Qkw03Fg1 | tr -d '\r' > /usr/bin/openshift-tests
```
I tried that at first but it became hard to digest so I'd rather keep it outside (you can't use multi-line cat in dockerfile so it'd have to be one long spaghetti)

wainersm replied:
Now I remember that I once tried to embedded a long script on dockerfile and I ran into the same problem. Ok, so let's use that approach of an intermediary script on pastebin

ldoktor commented:
@wainersm I fixed the `&&` issue and left the wrapper as a separate script and modified the link to point to the target location, which should result in missing file issue. Anyway the previous rehearsal jobs passed, results are available here:

* https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44073/rehearse-44073-periodic-ci-kata-containers-kata-containers-main-e2e-tests/1719299236090089472
* https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44073/rehearse-44073-periodic-ci-kata-containers-kata-containers-main-next-e2e-tests/1719299236169781248

wainersm commented:
/ph-rehearse ack

wainersm commented:
/lgtm

ldoktor commented:
/pj-rehearse skip
